submissions/machine_learning_model_code_testing_framework.md

```diff
@@ -0,0 +1,44 @@
+---
+title: "machine learning model code testing framework"
+date: 2023-08-11T00:00:00+10:00
+draft: true
+keywords:
+  - mlops
+  - machine learning models
+  - unit tests
+  - integration tests
+  - post-deployment tests
+tags:
+  - mlops
+author: "@the.engineer"
+# author_external_url : "https://github.com/wlogblog/"
+robots: index,follow
+---
+
+An important component to a successful machine learning model project is to have a solid automated testing framework. This is usually set up in the CI/CD pipeline and triggered every code push. As the code base gets larger, more complex and with more contributors adding to the code base, it will become a crucial part of your code development process. Although it may take some time initially setting it up, it will allow you to spend less time fixing code bugs in the long run.
+
+The key word here is 'code testing' which is aimed to test the code functionality. It's not model performance testing - this should be separate.
+
+Three basic tests to have in a mature machine learning model project are unit tests, integration tests and post-deployment tests.
+
+### 1. Unit tests
+Unit tests are designed to test individual components of the code base. It will test that the function accepts the same input and output and will run without breaking. Unit tests allows you to easily identify which component of the code will break if some code changes. If the change was intentional, it's just a matter of updating the unit test.
```

Comment:
Could you add some sample tooling for unit testing? For example, maybe `pytest` for python projects. 

Also, what kind of stuff would require unit testing? Is it necessary to mock an s3 call in an ML project? etc.